storage/series/buffer.go

```diff
@@ -365,15 +367,28 @@ func (b *dbBuffer) Snapshot(ctx context.Context, blockStart time.Time) [][]xio.S
 			return
 		}
 
+		// We need to merge all the bootstrapped blocks / encoders into a single stream for the sake
+		// of being able to persist it to disk as a single encoded stream.
+		bucket.merge()
```

Comment:
Can you deal with the error case perhaps?  i.e. log an error like we do in the buffer tick method?

Reply:
I actually refactored the Snapshot method to return an error so I can just do that